server/api/routes/service.ts

```diff
@@ -0,0 +1,43 @@
+import * as mongoose from "mongoose";
+import * as jwt from "jsonwebtoken";
+import { UserModel } from "../models/user";
+import { HeaderMiddleware } from "../middlewares/header";
+import { ServiceController } from "../controllers/service";
+import { Request, Response, NextFunction } from "express";
+
+const User = mongoose.model("User", UserModel);
+
+export class ServiceRoutes {
+  public service: ServiceController = new ServiceController();
+
+  public routes(app): void {
+    app.use((req: Request, res: Response, next: NextFunction) => {
+      if (req.headers["authorization"]) {
+        jwt.verify(
+          req.headers["authorization"],
+          process.env.JWT_SECRET,
+          async (err, decoded) => {
+            if (err) {
+              return res.status(403).send({
+                success: false,
+                message: "Invalid token",
+              });
+            } else {
+              let profile = await User.findOne({ mobile: decoded.mobile });
```

Comment:
I think it should be email here, the User Model doesn't have a `mobile` column as of now.

Reply:
Oh! The service discovery endpoint was working even after this error, because it picked the first user as a fallback, and I was the first user in the db. What a coincidence. 😆 

Anyway, it was super important to catch this. Good eye! @utkarsh-dixit

Reply:
I can still see `mobile` here, re-opening the conversation again